src/Phase/Phase.cs

```diff
@@ -16,14 +16,14 @@ public sealed class Phase : ITenantContext
         private readonly SemaphoreSlim _lock = new SemaphoreSlim(1, 1);
         private PhaseState _currentState = new VacantState();
 
-        internal Phase(DependencyResolver resolver, IEventsProvider eventsProvider, Func<string, IDictionary<string, string>> tenantKeysFactory)
+        internal Phase(DependencyResolver resolver, IEventsProvider eventsProvider, Func<string, IDictionary<string, string>> tenantKeysFactory, EventPublisher publisher = null)
```

Comment:
Optional params on a constructor 😮 
Make a contract with a consumer. Tell them what you need to do your job, and don't be wishy washy about it :) Once you put it out there as something the consumer of this class _can_ control, it immediately becomes something the consumer _should_ control.

Reply:
Im guessing that you are leaning towards making EventPublisher a required parameter rather than optional. We can control whats injected through the fluent builders. Default ones inject the standard EventPublisher and the testing ones inject the new one we are adding with this PR.

Ill go ahead and make this change but, I think what we do here kind of depends on what we do about your first comment. Let me know if you have any specific suggestions